Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/AzureCliAzureManager.java

```diff
@@ -0,0 +1,236 @@
+    private static final String FAILED_TO_AUTH_WITH_AZURE_CLI = "Failed to auth with Azure CLI";
+    private static final String UNABLE_TO_GET_AZURE_CLI_CREDENTIALS = "Unable to get Azure CLI credentials, " +
+            "please ensure you have installed Azure CLI and signed in.";
+
+    private static Settings settings;
+    private Azure.Authenticated authenticated;
+    private AzureCliCredentials azureCliCredentials;
+    private SubscriptionManager subscriptionManager;
+
+    static {
+        settings = new Settings();
+        settings.setSubscriptionsDetailsFileName(FILE_NAME_SUBSCRIPTIONS_DETAILS_AZ);
+    }
+
+    private static class LazyLoader {
+        static final AzureCliAzureManager INSTANCE = new AzureCliAzureManager();
+    }
+
+    public static class AzureCliAzureManagerFactory implements AzureManagerFactory {
+
+        @Override
+        public @Nullable AzureManager factory(AuthMethodDetails authMethodDetails) {
+            return getInstance().isSignedIn() ? getInstance() : null;
+        }
+
+        @Override
+        public AuthMethodDetails restore(final AuthMethodDetails authMethodDetails) {
+            try {
+                getInstance().signIn();
+            } catch (AzureExecutionException e) {
+                // Catch the exception when restore
+            } finally {
+                return authMethodDetails;
+            }
+        }
+    }
+
+    public static AzureCliAzureManager getInstance() {
+        return LazyLoader.INSTANCE;
+    }
+
+    @Override
+    public Azure getAzure(String sid) throws IOException {
+        if (sidToAzureMap.containsKey(sid)) {
+            return sidToAzureMap.get(sid);
+        }
+        final Azure azure = auth().withSubscription(sid);
+        // TODO: remove this call after Azure SDK properly implements handling of unregistered provider namespaces
+        AzureRegisterProviderNamespaces.registerAzureNamespaces(azure);
+        sidToAzureMap.put(sid, azure);
+        return azure;
+    }
+
+    @Override
+    public AppPlatformManager getAzureSpringCloudClient(String sid) {
+        return sidToAzureSpringCloudManagerMap.computeIfAbsent(sid, s ->
+                buildAzureManager(AppPlatformManager.configure()).authenticate(azureCliCredentials, s));
+    }
+
+    @Override
+    public InsightsManager getInsightsManager(String sid) {
+        return sidToInsightsManagerMap.computeIfAbsent(sid, s ->
+                buildAzureManager(InsightsManager.configure()).authenticate(azureCliCredentials, s));
+    }
+
+    @Override
+    public List<Subscription> getSubscriptions() {
+        return authenticated.subscriptions().list();
+    }
+
+    @Override
+    public List<Pair<Subscription, Tenant>> getSubscriptionsWithTenant() {
+        final Tenant subscriptionTenant = authenticated.tenants().list().stream()
+                .filter(tenant -> StringUtils.equals(tenant.tenantId(), authenticated.tenantId()))
+                .findFirst().orElse(null);
```

Comment:
I don't think we should return a List(Pair(sid, null)) here. Let's return an empty list for tenant not found.